barretenberg/cpp/src/barretenberg/proof_system/op_queue/ecc_op_queue.hpp

```diff
@@ -274,6 +255,31 @@ class ECCOpQueue {
             .mul_scalar_full = 0,
         });
     }
+
+  private:
+    /**
+     * @brief TESTING PURPOSES ONLY: Populate ECC op queue with mock data as stand in for "previous circuit" in tests
+     * @details TODO(#723): We currently cannot support Goblin proofs (specifically, transcript aggregation) if there
+     * is not existing data in the ECC op queue (since this leads to zero-commitment issues). This method populates the
+     * op queue with mock data so that the prover of an arbitrary 'first' circuit can behave as if it were not the
+     * prover over the first circuit in the stack. This method should be removed entirely once this is resolved.
+     *
+     * @param op_queue
+     */
+    void populate_with_mock_initital_data()
```

Comment:
I'm actually a bit confused about why this is working. There were previously two versions of this "mocked op queue" method. The older one looked like this one, where we add nonsense data. This was implemented before ECCVM/Translator were finished. The newer one was similar but added valid ops (not random ones) to the op queue. The method you have here should cause the ECCVM to fail... I don't understand why it doesn't

Reply:
I did notice the other one was computing actual commitments, will make sure we continue doing that in this method and investigate ECCVM then

Reply:
The commitments are valid in both cases, its the underlying group operations themselves that are valid in the newer method but just random garbage in this method. I wonder if somehow the first row of the eccvm is getting ignored due to some zero shift quirk or something..